### PR TITLE
Implement Dir(Iter) and each(Dir)

### PR DESCRIPTION
**Pull request by antonio-pedro99**
Solving issue #475 

I did not find a good test cases as direcories might be subjective.

### CODE REVIEW

**Comment by antonio-pedro99** [comment]
@ilyash this is for #601 issue.

File: lib/autoload/globals/Iter.ngs [hunk 1]
```diff
@@ -373,3 +373,48 @@ F skip(i:Iter, count:Int){
...
+
+# ---------- DirIter ----------
+doc Calls DirIter constructor.
+doc %RET - DirIter which allows us to iterate over a directory
+F Iter(d:Dir) DirIter(d)
+
+doc Initialize the DirIter with specific Dir
+F init(i:DirIter, d:Dir){
+	i.val = d
+	i.paths = dir(d)
```
Comment (ilyash-b) [inline lib/autoload/globals/Iter.ngs:388]: Don't have much time now but this is surely a big issue. Don't load everything at once into memory. The iterator should be smarter than that. It should use `c_readdir()` as necessary. Think about `Dir(...).Iter().limit(5)`.

File: lib/autoload/globals/Iter.ngs [hunk 2]
```diff
@@ -373,3 +373,48 @@ F skip(i:Iter, count:Int){
...
+
+doc Calls cb with each path from the directory
+doc %RET - i
+F each(d:Dir, cb:Fun){
+	i = DirIter(d)
+	for e in i {
+		cb(e)
+	}
+	i
+}
```
Comment (ilyash-b) [inline lib/autoload/globals/Iter.ngs:420]: Should reuse existing `each(i:Iter, cb:Fun)`.

So something like `F each(d:Dir, cb:Fun) d.Iter().each(cb)` (untested)

File: lib/autoload/globals/Iter.ngs [hunk 3]
```diff
@@ -373,3 +373,48 @@ F skip(i:Iter, count:Int){
...
+	d.idx += 1
+	ret
+}
+
+doc Method that allows us to iterate the specified Dir()
+doc %RET - DirIter
+F Iter(d:Dir){
+	i = DirIter(d)
+	i
+}
```
Comment (ilyash-b) [inline lib/autoload/globals/Iter.ngs:410]: Duplicate of `Iter(Dir)` defined above

**Review by ilyash-b — CHANGES_REQUESTED** [review]

**Review by ilyash-b — CHANGES_REQUESTED** [review]
PR now includes completely unrelated changes

**Comment by antonio-pedro99** [comment]
I commited a mistake, I forgot to checkout to the correct branch.

File: lib/autoload/globals/Iter.ngs (hunk 1 above)
Comment (antonio-pedro99) [reply lib/autoload/globals/Iter.ngs:388]: I do not know if I understood the point completly. 
But, taking look at F dir(...) in stdlibs.ngs, I could see that it is using both c_readdir() and c_opendir(). Would not be the same using c_readdir directly for loading in this method?

**Comment by ilyash-b** [comment]
> I commited a mistake, I forgot to checkout to the correct branch.

It's not clear whether you intend to clean this up

**Comment by antonio-pedro99** [comment]
I have canceled the last commit due to the previous mistake. 

I have pushed a clean commit without unrelated changes.

I am still working on the changes you asked to make.

File: lib/autoload/globals/Iter.ngs (hunk 2 above)
Comment (antonio-pedro99) [reply lib/autoload/globals/Iter.ngs]: is this correct?

File: lib/autoload/globals/Iter.ngs (hunk 1 above)
Comment (ilyash-b) [reply lib/autoload/globals/Iter.ngs:388]: `c_readdir()` must be used in all cases. The difference is when and how many times it is called. If you need to return an `Arr` you must read the whole directory in any case. If you are returning an `Iter`, `c_readdir()` should be called when necessary, that is when `next()` is called. There is no point reading the whole directory. Imagine `next()` will be called just one time or not at all.

**Pull request closed without merging**